### DISCUSSION TITLE
Simple image markdown output

### DISCUSSION THREAD

**Discussion by gcalinBR** [pawamoy/markdown-exec#46] [Q&A]
Hi everyone,
Just started testing matkdown-exec and I didn't manage to output a simple image reference:

```
` ``python exec="true"
print("![ALT TEXT](image.png)")
` ``

```
I believe it should output "!\[ALT TEXT\](image.png)" and markdown would process it normally. Instead it just outputs "ALT TEXT".
Could someone point my mistake?
Thanks
gcalin

**Comment by pawamoy** [comment]
Can you inspect the HTML page and paste the resulting HTML here? Is it possible that your link to the image is broken and therefore only the alt text appears?

**Reply by gcalinBR** [reply]
Just found out. For some reason the path is processed slightly different when the image markup is outputted from the python code. When the markup is placed directly in the .md, the markup ! [ALT TEXT] (image.png) outputs "../image.png" in the HTML. When it is outputted from the python code, it just outputs "image.png" in the HTML, breaking the link.

I changed the print in the python code to print("! [ALT TEXT] (../image.png)") and it worked, however I wonder why it is being processed different. As I understand the python output is feed directly in the .md file during the preprocessor, correct? If so, it should process exactly as a markup directly typed in the md file.

Thank you for pointing the direction.
Gabriel

**Reply by pawamoy** [reply]
If you can provide a reproducible I will investigate. Currently I'm not able to experience the behavior you describe:

````md
```python exec="true"
print("![ALT TEXT](logo.svg)")
```

![ALT TEXT](logo.svg)
````

...produces this HTML:

```html
<p><img alt="ALT TEXT" src="logo.svg"></p>
<p><img alt="ALT TEXT" src="logo.svg"></p>
```

**Reply by gcalinBR** [reply]
That was exactly my test:

````
# Works ok
![ALT TEXT](image.png)

# Do not work
```python exec="true"
print("![ALT TEXT](image.png)")
```

````

.. produces this HTML on my end:

````
<h1 id="works-ok">Works ok</h1>
<p><img alt="ALT TEXT" src="../image.png" /></p>
<h1 id="do-not-work">Do not work</h1>
<p><img alt="ALT TEXT" src="image.png" /></p>
````

not sure why mkdocs generate the first instance with "../" in the file path
